Assets/_BForBoss/_Weapons/Scripts/WeaponBehaviour.cs

```diff
@@ -43,25 +43,27 @@ public void Initialize(IWeaponProperties properties = null)
 
         private void BindWeapon()
         {
-            _weapon.OnFireWeapon += Fire;
+            _weapon.OnFireWeapon += HandleOnFire;
         }
         
-        protected abstract void OnFire(InputAction.CallbackContext context);
+        protected abstract void OnFireInputAction(InputAction.CallbackContext context);
         protected abstract void Update();
         
-        private void Fire()
+        private void HandleOnFire(int numberOfBullets)
         {
-            GenerateBullet(_firePoint.position, GetDirectionOfShot());
+            for (int i = 0; i < numberOfBullets; i++)
+            {
+                GenerateBullet(_firePoint.position, GetDirectionOfShot());
+            }
         }
         
         //Placeholder
         protected void GenerateBullet(Vector3 position, Vector3 fireDirection)
         {
             var bullet = GameObject.CreatePrimitive(PrimitiveType.Sphere);
-            bullet.transform.localScale = new Vector3(0.2f, 0.2f, 0.2f);
+            bullet.transform.localScale = new Vector3(0.05f, 0.05f, 0.05f);
```

Comment:
maybe make vector a readonly? looks like magic numbers

Reply:
This is whole method placeholder code and can be completely removed/changed.